engine/language-server/src/test/scala/org/enso/languageserver/buffer/RopeSpecification.scala

```diff
@@ -0,0 +1,163 @@
+package org.enso.languageserver.buffer
+import org.enso.languageserver.data.buffer.Rope
+import org.scalacheck.Prop.forAll
+import org.scalacheck.Arbitrary._
+import org.scalacheck.Properties
+
+object RopeSpecification extends Properties("Rope") {
+
+  private def ropeFromStrings(strings: List[String]): Rope =
+    strings.foldLeft(Rope.empty)((rope, string) => rope ++ Rope(string))
+
+  private def normalizeOffset(offset: Int, length: Int): Int =
+    (offset % (length + 1)).abs
+
+  private def normalizeIndex(index: Int, length: Int): Int =
+    (index % length).abs
+
+  // Rope
```

Comment:
Can you use the usual section dividers?
```suggestion
  // === Rope=================================================================
```